packages/runtime-core/src/componentOptions.ts

```diff
@@ -630,6 +630,14 @@ export function applyOptions(
       ? provideOptions.call(publicThis)
       : provideOptions
     for (const key in provides) {
+      //#2306
+      if (
+        instance.provides &&
+        (key in instance.provides) &&
+        asMixin
+      ) {
+        continue
+      }
```

Comment:
I guess we should still set the value to `instance.provides` and not break the prototype chain, so that
- The instance's children can inherit the new `provides` from direct parent